### PR TITLE
Use keyserver that still returns keys with user IDs

### LINKED ISSUE #2050: [BUG] unstable build dependending on keys.gnupg.net
# Bug Report

## Context

Building the image on our CI fails some days when keys.gnupg.net doesn't return the key for 0x683BF1BEBD0A882C

### What is affected by this bug?

Building the image

### When does this occur?

Sometimes (e.g. today, 2021-06-22), it also happened a few weeks ago but was resolved during the day

### How do we replicate the issue?

1. Run `gpg --keyserver keys.gnupg.net --recv-keys "0x683BF1BEBD0A882C"`

### Actual Behavior
gpg just returns 
`gpg: keyserver receive failed: No name`

### Expected Behavior
gpg should receive the key

## Your Environment

- version: `master`
- available RAM: `128GB`
- Docker version: `20.10.7`
- docker-compose version: `1.29.2`

### PR DESCRIPTION
**Pull request by ap-wtioit**
# Description

Use keyserver that still returns keys with user IDs

keys.gnupg.net doesn't return user IDs (without approval) and therefore gpg
doesn't accept keys from it.

Fixes #2050 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

FYI @wt-io-it

### CODE REVIEW

**Review by wernerfred — APPROVED** [review]
LGTM if the pipeline passes.

Thanks for your contribution and the very detailed description of your approach in the linked Issue. Helps a lot keeping track and follow along with your thoughts.

We used keys.gnupg.net as the fail2ban maintainers referenced this server when we asked for their public key. Good finding that you discovered a (only sometimes appearing?) bug in this implementation.

Thanks

**Review by georglauterbach — APPROVED** [review]
Good catch!

**Comment by casperklein** [comment]
> keys.gnupg.net doesn't return user IDs (without approval)

I stumbled upon the same yesterday and hoped that was only temporary. For completeness: It's not about a missing feature, the host doesn't even resolve to an IP address anymore.

**Comment by casperklein** [comment]
I just found out, that the DNS record for the server pool was intentionally removed.

> 
> This service is deprecated. This means it is no longer maintained, and new HKPS certificates will not be issued. Service reliability should not be expected.
> 
> Update 2021-06-21: Due to even more GDPR takedown requests, the DNS records for the pool will no longer be provided at all.

Source: https://sks-keyservers.net/status/

`keys.gnupg.net` was a CNAME for `hkps.pool.sks-keyservers.net`

**Pull request merged**
